pkg/registry/node/etcd/etcd.go

```diff
@@ -102,3 +103,25 @@ var _ = rest.Redirector(&REST{})
 func (r *REST) ResourceLocation(ctx api.Context, id string) (*url.URL, http.RoundTripper, error) {
 	return node.ResourceLocation(r, r.connection, r.proxyTransport, ctx, id)
 }
+
+// An interface for types that allow getting information about a Node on which give pod is running.
+type HostLocator interface {
```

Comment:
I don't see KubeletClient being used anywhere outside of pod and node storage construction. Can we just rename it to KubeletClientDefaults, pass it into node/etcd#NewREST and get back another ConnectionInfoGetter that resolves node-specific info? I think that signature would change to:

`func NewREST(s storage.Interface, useCacher bool, kubeletClientDefaults client.ConnectionInfoGetter, proxyTransport http.RoundTripper) (*REST, *StatusREST, ConnectionInfoGetter)`